v2/rust/src/scheduling.rs

```diff
@@ -0,0 +1,175 @@
+use super::attempt::{Attempt, Identifier, RetrySpec};
+use super::config::{Config, SuiteConfig};
+use super::environment::{Environment, ResultCode};
+use super::logging::log_and_return_error;
+use super::results::{suite_result_file, suite_results_directory};
+use super::session::{RunOutcome, Session};
+use super::termination::TerminationFlag;
+
+use anyhow::{bail, Context, Result};
+use chrono::Utc;
+use clokwerk::{Scheduler, TimeUnits};
+use log::{debug, error};
+use std::fs::create_dir_all;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex, MutexGuard, TryLockError};
+use std::thread::{sleep, spawn};
+use std::time::Duration;
+
+pub fn run_suites(config: &Config, termination_flag: &TerminationFlag) -> Result<()> {
+    let mut scheduler = Scheduler::new();
+
+    for (suite_name, suite_config) in config.suites() {
+        let suite_run_spec = Arc::new(SuiteRunSpec {
+            termination_flag: termination_flag.clone(),
+            parallelism_protection: Mutex::new(0),
+            suite_name: suite_name.clone(),
+            suite_config: suite_config.clone(),
+            working_directory: config.working_directory.clone(),
+            result_file: suite_result_file(
+                &suite_results_directory(&config.results_directory),
+                suite_name,
+            ),
+        });
+        scheduler
+            .every(
+                suite_config
+                    .execution_config
+                    .execution_interval_seconds
+                    .seconds(),
+            )
+            .run(move || run_suite_in_new_thread(suite_run_spec.clone()));
+    }
+
+    loop {
```

Comment:
OTI: We could think about using a different scheduler here. I'll talk to Robert about the discussions he opened.